src/main/java/com/allenai/ml/linalg/SparseVector.java

```diff
@@ -87,5 +89,5 @@ public double dotProduct(Vector other) {
           result += val * other.at(idx);
```

Comment:
Yep, in Mahout we did exactly like that, advancing iterators (as long as
they were sorted) in parallel, and with no object creation, but just
forwarding Element entries like you've got here.

On Tue, May 12, 2015 at 10:46 PM, Aria Haghighi <notifications@github.com>
wrote:

> In src/main/java/com/allenai/ml/linalg/SparseVector.java
> <https://github.com/allenai/ml/pull/6#discussion_r30203876>:
>
> > @@ -87,5 +89,5 @@ public double dotProduct(Vector other) {
> >            result += val * other.at(idx);
>
> Ah, yes you're right I've written code like the above before. I did it
> access to an iterator (like Vector.Iterator) rather than the direct
> arrays; I think you can still do the above efficiently, since you're only
> relying on advancing each of the cursors and retriving the index and value.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/allenai/ml/pull/6/files#r30203876>.
>



-- 

  -jake